notebooks/instructlab-knowledge/utils/illuminator/utils.py

```diff
@@ -50,41 +62,48 @@ def save_results(results, output_file: str) -> None:
         json.dump(results, f, indent=4)
     logger.info(f"📁 Results saved to {output_file}")
 
-def generate_summary(results) -> None:
+def generate_summary(results, file=None) -> None:
     """
-    Prints a human-readable summary of merged table cell issues per file.
+    Prints a human-readable summary of merged table cell issues per file,
+    optionally also writing to a file.
 
     Args:
-        results: Dictionary containing analysis results for each PDF.
+        results: Dictionary containing analysis results for each document.
+        file: Optional file-like object (e.g., open file or StringIO) to also write output.
     """
-    logger.info("📊 Summary Report")
-    logger.info("=" * 50)
-    for file, data in results.items():
-        logger.info(f"\n📂 File: {file}")
+    def out(msg):
+        print(msg)               # Always print to notebook
+        if file:
+            print(msg, file=file)  # Also write to file if provided
```

Comment:
This can be done with logging handlers: https://stackoverflow.com/questions/13733552/logger-configuration-to-log-to-file-and-print-to-stdout

That way, the logging configuration can be declared at the top of the file and this function isn't needed.

Reply:
I was previously using logger handlers, but I found print() was a "notebook friendlier" approach. 

logger.info() outputs to stderr, which shows up red even when it's not an error, confusing for notebook users. print() gave better control over the output destination and formatting

Reply:
Interesting. Makes sense then.